samtranslator/sdk/resource.py

```diff
@@ -30,7 +32,15 @@ def valid(self):
 
         :return: True, if the resource is valid
         """
-        # As long as the type is valid.
+        # As long as the type is valid and type string.
+        # validate the condition should be string
+
+        if self.condition:
+
+            if not isinstance(self.condition, string_types):
```

Comment:
Can we use `is_str()` instead? I'd just like to be consistent in how we do type checking across the codebase.